src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -58,21 +59,33 @@ struct list_offsets_ctx {
       , unauthorized_topics(std::move(unauthorized_topics)) {}
 };
 
+static std::optional<partition_proxy> make_partition_proxy(
+  const model::materialized_ntp& mntp,
+  ss::lw_shared_ptr<cluster::partition> partition,
+  cluster::partition_manager& pm) {
+    if (!mntp.is_materialized()) {
+        return make_partition_proxy<replicated_partition>(partition);
+    }
+    if (auto log = pm.log(mntp.input_ntp()); log) {
+        return make_partition_proxy<materialized_partition>(*log);
```

Comment:
it looks like you could put `make_materialized_partition_proxy` in `partition_proxy.h` and the definition of it in whatever .cc file has the implementation?